source/agora/consensus/EnrollmentManager.d

```diff
@@ -420,6 +420,37 @@ public class EnrollmentManager
         return this.validator_set.count();
     }
 
+    /***************************************************************************
+
+        Check if an enrollment is a valid candidate for the proposed height
+
+        Params:
+            enroll = The enrollment of the target to be checked
+            height = the height of proposed block
+            findUTXO = delegate to find the referenced unspent UTXOs with
+
+        Returns:
+            `null` if the enrollment can be a validator at the proposed height,
+            otherwise a string explaining the reason it is invalid.
+
+    ***************************************************************************/
+
+    public string isValidCandidate (const ref Enrollment enroll, Height height,
```

Comment:
This should be named `isInvalidCandidateReason`, not "isValid". Because you return a string when it's invalid, so `if (isValidCandidate(...))` will be `true` even though it failed validation.